sale_invoice_delivery_state/models/sale.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2023 Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.depends("delivery_state")
+    def _get_invoice_status(self):
+        super()._get_invoice_status()
+        for sale in self.filtered(lambda s: s.invoice_status == "to invoice"):
+            if (
+                sale.partner_id.invoice_policy == "fully"
+                and sale.delivery_state != "done"
+            ):
+                sale.invoice_status = "no"
```

Comment:
you could write in all the orders at once, no need to loop through all
```suggestion
        open_orders = self.filtered(
            lambda s: s.invoice_status == "to invoice"
            and s.partner_id.invoice_policy == "fully"
            and s.delivery_state != "done"
        )
        open_orders.write({"invoice_status": "no"})
```

Reply:
@gbrito , it's done.